internal/blocksync/reactor.go

```diff
@@ -184,21 +187,21 @@ func (r *Reactor) OnStop() {
 // respondToPeer loads a block and sends it to the requesting peer, if we have it.
 // Otherwise, we'll respond saying we do not have it.
 func (r *Reactor) respondToPeer(ctx context.Context, msg *bcproto.BlockRequest, peerID types.NodeID, blockSyncCh *p2p.Channel) error {
-	block := r.store.LoadBlock(msg.Height)
+	block := r.store.LoadBlockProto(msg.Height)
 	if block != nil {
 		extCommit := r.store.LoadBlockExtendedCommit(msg.Height)
 		if extCommit == nil {
 			return fmt.Errorf("found block in store without extended commit: %v", block)
 		}
-		blockProto, err := block.ToProto()
-		if err != nil {
-			return fmt.Errorf("failed to convert block to protobuf: %w", err)
-		}
+		// blockProto, err := block.ToProto()
```

Comment:
It seems you can remove this part of the code, since `block` is now a proto version.